packages/types/src/settings.ts

```diff
@@ -1,26 +1,152 @@
 import { Package } from ".";
 import { DeepPartial } from "ts-essentials";
 
-// Gets a Package and returns only what's needed for the frontity.settings.js file.
-type PackageForSettings<Pkg extends Package> = {
-  name: Required<Pkg["name"]>;
-  active?: boolean;
-  state?: DeepPartial<Pkg["state"]>;
-};
+/**
+ * Gets a Package and returns only what's needed for the frontity.settings.js file.
+ */
+type PackageForSettings<Pkg> = Pkg extends Package
+  ? {
+      /**
+       * Package name.
+       */
+      name: Required<Pkg["name"]>;
 
+      /**
+       * Flag indicating whether the package is active or not.
+       */
+      active?: boolean;
+
+      /**
+       * Package settings.
+       */
+      state?: DeepPartial<Pkg["state"]>;
+    }
+  : never;
```

Comment:
This is just a nit, but is there a reason why you removed the constraint on the generic? I think that the previous form like:
```ts
type PackageForSettings<Pkg extends Package> = { ... };
```

is generally  better than:

```ts
type PackageForSettings<Pkg> = Pkg extends Package ? { ... } : never;
```

In the first case, you should get a compiler error, if you pass something other than a Package as a type parameter right on that line: 

```ts
packages: (string | PackageForSettings<Pkg>)[];
```

In practice it doesn't matter much in this case because this type is only used internally

Reply:
The thing I wanted to do is to resolve each package independently.

When you define a `Settings` type, you would write something like the following:

```ts
const settings: Settings<
  | TinyRouter
  | WpSource
  | MarsTheme
  | HeadTags
> = [{ ... }];
```

As the package you are passing is actually a union of packages, the previous implementation returned the resolved state for the union of all packages, and therefore you would have the types of all packages in every package.

By using the conditional operator with the union of all packages, each package only contains its own settings as they are handled one by one.

You can take a look at `/e2e/project/frontity.settings.ts` and play a bit with types to check that.

Reply:
oh, I see, thanks for explaining!

Reply:
So, I guess that can do both in that case, correct?

```
type PackageForSettings<Pkg extends Package> = Pkg extends Package ? { ... } : never;
```

Reply:
I guess, haha. I'll do that. :blush: